crates/xline-client/src/types/kv.rs

```diff
@@ -1,29 +1,68 @@
-use xlineapi::command::KeyRange;
+use crate::error::Result;
+use curp::client::ClientApi;
+use futures::FutureExt;
+use pin_project::pin_project;
+use std::future::Future;
+use std::pin::Pin;
+use std::{pin::pin, sync::Arc};
+use tonic::Status;
+use xlineapi::{
+    command::{Command, CommandResponse, KeyRange, SyncResponse},
+    execute_error::ExecuteError,
+    RequestWrapper,
+};
 pub use xlineapi::{
     CompactionResponse, CompareResult, CompareTarget, DeleteRangeResponse, PutResponse,
     RangeResponse, Response, ResponseOp, SortOrder, SortTarget, TargetUnion, TxnResponse,
 };
 
-/// Request type for `Put`
-#[derive(Debug, PartialEq)]
-pub struct PutRequest {
-    /// Inner request
-    inner: xlineapi::PutRequest,
+/// The type of Response
+type ResponseType = std::result::Result<
+    std::result::Result<(CommandResponse, Option<SyncResponse>), ExecuteError>,
+    Status,
+>;
+
+/// The type of the future inside [`PutFut`].
+// type RequestFutureType<'a> = Option<BoxFuture<'a, ResponseType>>;
+type RequestFutureType = Option<Pin<Box<dyn Future<Output = ResponseType> + Send>>>;
+
+/// Future for `Put`, make it awaitable.
+///
+/// Before first poll, fut will be [`Option::None`].
+/// Once It's been polled, fut will be [`Option::Some`] and inner will be [`Option::None`].
+#[pin_project]
+pub struct PutFut {
```

Comment:
I think it's better to add a lifetime parameter to this struct. Otherwise it may outlive the original client.